core/state/journal.go

```diff
@@ -141,11 +161,13 @@ func (j *journal) copy() *journal {
 	for i := 0; i < j.length(); i++ {
 		entries = append(entries, j.entries[i].copy())
 	}
+
 	return &journal{
 		entries:        entries,
 		dirties:        maps.Clone(j.dirties),
 		validRevisions: slices.Clone(j.validRevisions),
 		nextRevisionId: j.nextRevisionId,
+		storageEntries: slices.Clone(j.storageEntries),
```

Comment:
the beheavior of storageChangePool is not right IMO.
after copying, a journal may still holding pointers in storageEntries which have been put back.